src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -439,5 +442,118 @@ public static string MapNameToOID(string name)
 
             return oidName;
         }
+
+        public static byte[] EncodeOID(string str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+
+            string[] oidString = str.Split(SepArray);
+            uint[] oidNums = new uint[oidString.Length];
+            for (int i = 0; i < oidString.Length; i++)
+            {
+                oidNums[i] = (uint)int.Parse(oidString[i], CultureInfo.InvariantCulture);
+            }
+
+            // Handle the first two oidNums special
+            if (oidNums.Length < 2)
+                throw new CryptographicUnexpectedOperationException(SR.Cryptography_InvalidOID);
+
+            uint firstTwoOidNums = (oidNums[0] * 40) + oidNums[1];
+
+            // Determine length of output array
+            int encodedOidNumsLength = 2; // Reserve first two bytes for later
+            encodedOidNumsLength += GetEncodedOidLength(firstTwoOidNums);
+            for (int i = 2; i < oidNums.Length; i++)
+            {
+                encodedOidNumsLength += GetEncodedOidLength(oidNums[i]);
+            }
+
+            // Allocate the array to receive encoded oidNums
+            byte[] encodedOidNums = new byte[encodedOidNumsLength];
+            int encodedOidNumsIndex = 2;
+
+            // Encode each segment
+            EncodeSingleOidNum(firstTwoOidNums, encodedOidNums, ref encodedOidNumsIndex);
+            for (int i = 2; i < oidNums.Length; i++)
+            {
+                EncodeSingleOidNum(oidNums[i], encodedOidNums, ref encodedOidNumsIndex);
+            }
+
+            // Final return value is 06 <length> encodedOidNums[]
+            if (encodedOidNumsIndex - 2 > 0x7f)
+                throw new CryptographicUnexpectedOperationException(SR.Cryptography_Config_EncodedOIDError);
+
+            encodedOidNums[0] = (byte)0x06;
+            encodedOidNums[1] = (byte)(encodedOidNumsIndex - 2);
+
+            return encodedOidNums;
+        }
+
+        private static void EncodeSingleOidNum(uint value, byte[] destination, ref int index)
+        {
+            // Write directly to destination starting at index, and update index based on how many bytes written.
+            if (checked((int)value) < 0x80)
+            {
+                destination[index++] = (byte)value;
+                Debug.Assert(GetEncodedOidLength(value) == 1);
```

Comment:
Instead of having two different methods, how about something like:

```C#
if (destination == null)
{
    index += 1;
}
else
{
    destination[index++] = (byte)value;
}
```

repeat pattern for other cases.  Then assert at the end of the non-null call that index is equal to the length computed from the first pass.